project-base/app/src/Component/FileUpload/FileUpload.php

```diff
@@ -79,11 +78,15 @@ private function getPositionForNewEntity(EntityFileUploadInterface $entity): int
         $entityId = $entity->getEntityId();
         $type = $entity->getType();
         $uploadEntityType = $this->getUploadEntityType($entity);
+        $key = sprintf('%s~%s~%s~%s', $entityName, $entityId, $type, $uploadEntityType);
```

Comment:
are you going to use that `getKeyFromParts` here?
We can discuss the interface unification yet - some public methods of LocalCacheFacade now accept the key itself, while some expect the key parts and build the key themselves. Maybe all the public methods should accept just the key parts so you do not have to build the key yourself?